linchpin/shell/__init__.py

```diff
@@ -91,12 +89,17 @@ def _handle_results(ctx, results, return_code):
 
     # PRINT OUTPUT RESULTS HERE
     ctx.log_state(output)
+    distill_on_error = json.loads(lpcli.get_cfg('lp',
+                                                'distill_on_error').lower())
 
-#   FIXME: have use_actual_rcs be a flag
+    #   FIXME: have use_actual_rcs be a flag
     use_actual_rcs = True
-    if use_actual_rcs:
+    if use_actual_rcs and not(distill_on_error):
```

Comment:
I see you do have 'use_actual_rcs', but that should come from the linchpin.constants (or overridden by the linchpin.conf) and set to a default. Not hard coded here. Additionally, distill_on_error is a separate function and should occur if and only if that flag is set, it shouldn't be dependent on whether 'use_actual_rcs' is set. This would mean that one is dependent on the other to be true for certain conditions.